src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/AutoInsert/AutoClosingTagOnAutoInsertProvider.cs

```diff
@@ -234,9 +234,25 @@ private static bool TryEnsureOwner_WorkaroundCompilerQuirks(int afterCloseAngleI
                 //
                 // The owner will be the </div>. Note this does not happen outside of C# blocks.
 
-                var closeAngleIndex = afterCloseAngleIndex - 1;
-                var closeAngleSourceChange = new SourceChange(closeAngleIndex, length: 0, newText: string.Empty);
+                var closeAngleSourceChange = new SourceChange(afterCloseAngleIndex - 1, length: 0, newText: string.Empty);
                 currentOwner = syntaxTree.Root.LocateOwner(closeAngleSourceChange);
+
+                // Get the real closing angle if we get the quote from an attribute syntax. See https://github.com/dotnet/razor-tooling/issues/5694
+                switch (currentOwner)
+                {
+                    case MarkupTextLiteralSyntax { Parent: MarkupAttributeBlockSyntax { Parent: MarkupStartTagSyntax startTag } }:
```

Comment:
For all the conditions in the switch block is restricting the attributes parent to be a start tag / start taghelper required?

Reply:
Not sure. I'll check this

Reply:
`MarkupStartTagSyntax` and `MarkupTagHelperStartTagSyntax` don't seem to have an ancestor with `CloseAngle` property. So, I think, This check is required here